.github/workflows/release-build.yml

```diff
@@ -0,0 +1,40 @@
+name: Release Build
+
+on:
+  push:
+    paths:
+      - 'CHANGELOG.md'
+    branches:
+      - master
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.7
+          bundler-cache: true
+      - run: bundle exec rake test
+      - name: Extract version from CHANGELOG.md
+        run: echo "VERSION=$(echo $(head -n 1 CHANGELOG.md) | awk -F ' ' '{print $1}')" >> $GITHUB_ENV
+      - name: Extract date from CHANGELOG.md
+        run: echo "DATE=$(echo $(head -n 1 CHANGELOG.md) | awk -F '[()]' '{print $2}')" >> $GITHUB_ENV
+      - name: Remove dots for VERSION for use in release notes
+        run: echo "VERSION_NO_DOTS=$(echo $VERSION | sed s/\\.//g)" >> $GITHUB_ENV
+      - name: Build gem
+        run: gem build aws-xray-sdk.gemspec
+      # Implicitly verifies that CHANGELOG and version.rb match
+      - name: Push gem
+        run: gem push aws-xray-sdk-${{ env.VERSION }}.gem
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1
```

Comment:
Not sure if you're aware but `create-release` is, for some awful reason, now deprecated and no longer supported by the GitHub team. They provide some alternatives here: https://github.com/actions/create-release#github-action---releases-api

I can't imagine there will be too many problems with using an unmaintained action like this since it really just wraps a GitHub API call, but probably worth choosing a supported action regardless.

Reply:
Yeah I noticed, it's unfortunate - for now still prefer to use a GitHub action instead of third-party for security reasons and since presumably it does continue to work for this simple one.